src/backend/distributed/shared_library_init.c

```diff
@@ -188,6 +196,32 @@ _PG_init(void)
 
 
 /*
+ * multi_log_hook intercepts postgres log commands. We use this to show the user a
+ * meaningful error message when a process receives a SIGINT from the distributed
+ * deadlock detector.
+ */
+static void
+multi_log_hook(ErrorData *edata)
+{
+	if (edata->elevel == ERROR && edata->sqlerrcode == ERRCODE_QUERY_CANCELED &&
+		MyBackendGotKilledDueToDeadlock())
```

Comment:
Well, it formally didn't get killed yet, but ;). And it's also not really killing, more cancelling - the session continues to live afterwards after all?